src/wasm/wasm-binary.cpp

```diff
@@ -107,10 +107,18 @@ void WasmBinaryWriter::writeResizableLimits(
                    (shared ? (uint32_t)BinaryConsts::IsShared : 0U) |
                    (is64 ? (uint32_t)BinaryConsts::Is64 : 0U);
   o << U32LEB(flags);
-  o << U32LEB(initial);
-  if (hasMaximum) {
-    o << U32LEB(maximum);
+  if (is64) {
+    o << U64LEB(initial);
+    if (hasMaximum) {
+      o << U64LEB(maximum);
+    }
+  } else {
+    o << U32LEB(initial);
+    if (hasMaximum) {
+      o << U32LEB(maximum);
```

Comment:
If this is common perhaps we could add a `UMemoryLEB()` etc., but if it's just here then maybe not worth it.

Reply:
searching for `U64LEB` this is so far the only location that is conditional on Memory64.